components/BottomNavigation/src/private/MDCBottomNavigationItemView.m

```diff
@@ -0,0 +1,339 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCBottomNavigationItemView.h"
+
+#import "MaterialRTL.h"
+#import "MaterialBottomNavigationStrings.h"
+#import "MaterialBottomNavigationStrings_table.h"
+#import "MDCBottomNavigationItemBadge.h"
+
+static const CGFloat kMDCBottomNavigationItemViewCircleLayerOffset = -6.f;
+static const CGFloat kMDCBottomNavigationItemViewCircleLayerDimension = 36.f;
+static const CGFloat kMDCBottomNavigationItemViewCircleOpacity = 0.150f;
+static const CGFloat kMDCBottomNavigationItemViewTitleFontSize = 12.f;
+
+// The duration of the selection transition animation.
+static const NSTimeInterval kMDCBottomNavigationItemViewTransitionDuration = 0.180f;
+
+// The Bundle for string resources.
+static NSString *const kMaterialBottomNavigationBundle = @"MaterialBottomNavigation.bundle";
+static NSString *const kMDCBottomNavigationItemViewTabString = @"tab";
+
+@interface MDCBottomNavigationItemView ()
+
+@property(nonatomic, strong) CAShapeLayer *circleLayer;
+@property(nonatomic, strong) MDCBottomNavigationItemBadge *badge;
+@property(nonatomic, strong) UIImageView *iconImageView;
+@property(nonatomic, strong) UILabel *label;
+
+@end
+
+@implementation MDCBottomNavigationItemView
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCBottomNavigationItemViewInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self commonMDCBottomNavigationItemViewInit];
+  }
+  return self;
+}
+
+- (void)commonMDCBottomNavigationItemViewInit {
+  self.autoresizingMask = (UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight);
```

Comment:
This is set based on whether the view was build programmatically or in storyboards.  Do we need to override?

Reply:
Removed. The autoresizing masks don't seem to be necessary.